crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -271,6 +271,77 @@ impl<'w> EntityMut<'w> {
         self
     }
 
+    /// Inserts a component with the given `value`. Will replace the value if it already existed.
+    ///
+    /// **You should prefer to use the typed API [`EntityMut::insert`] where possible and only
+    /// use this in cases where there isn't a Rust type corresponding to the [`ComponentId`].
+    ///
+    /// # Safety
+    /// The value referenced by `value` must be valid for the given [`ComponentId`] of this world
+    pub unsafe fn insert_by_id(
+        &mut self,
+        component_id: ComponentId,
+        value: OwningPtr<'_>,
+    ) -> &mut Self {
+        // SAFETY: the caller promisees that `value` is valid for the `component_id`
+        self.insert_bundle_by_ids(vec![component_id], std::iter::once(value))
+    }
+
+    /// Inserts a bundle of components into the entity. Will replace the values if they already existed.
+    ///
+    /// **You should prefer to use the typed API [`EntityMut::insert_bundle`] where possible and only
+    /// use this in cases where there are no Rust types corresponding to the [`ComponentId`]s.
+    ///
+    /// # Safety
+    /// - each value of `components` must be valid for the [`ComponentId`] at the matching position in `component_ids` in this world
+    pub unsafe fn insert_bundle_by_ids<'a, I: IntoIterator<Item = OwningPtr<'a>>>(
+        &mut self,
+        mut component_ids: Vec<ComponentId>,
+        components: I,
+    ) -> &mut Self {
+        component_ids.sort();
```

Comment:
This is not obvious correct.

Do you not also need to simultaneously sort the `components`?